src/routers/api.js

```diff
@@ -305,6 +305,82 @@ router.post(
 	})
 );
 
+router.get(
+  '/tax-payers',
+  wrap(async (req, res) => {
+    try {
+      const api = await apiPromise;
+      const lastHeader = await api.rpc.chain.getHeader();
+			const politics = await api.query.llm.llmPolitics.entries();
+      const lastBlock = lastHeader.number.toNumber();
+
+      const limit = parseInt(req.query.limit, 10) || 10;
+      const months = parseInt(req.query.months, 10) || 12;
+      const blocksInDay = (3600 * 24) / 6;
+      const blocksInMonth = blocksInDay * 30.44;
+      const startBlock = Math.floor(lastBlock - (months * blocksInMonth));
+
+      const tax = await getTaxList();
+
+			const filteredData = tax.taxPools.nodes.reduce(
+				(acc, { addressId, value, blockNumber }) => {
+					const numericValue = Number(value);
+
+					if (blockNumber >= startBlock) {
```

Comment:
This seems to me to be risky over time, as you'll have to process more and more data on the server. Can't the query be optimized with date param?